client/src/components/Modal.jsx

```diff
@@ -0,0 +1,42 @@
+import { useRef } from "react";
+import "./Modal.css";
+import PropTypes from "prop-types";
+
+function Modal({ show, onClose, children }) {
+  const modalRef = useRef(null);
+
+  const handleBlur = (event) => {
+    if (!modalRef.current.contains(event.relatedTarget)) {
+      onClose();
+    }
+  };
+
+  if (!show) {
```

Comment:
Any reason not to do conditional rendering in the parent instead of this?

Reply:
OK. I tried this :

Modal.jsx:

 return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [show, onClose]);

  return (
    show && (
      <div className="modalOverlay" role="dialog" aria-modal="true">
.
.

Questionnaire.jsx:

        {showModal && (
          <Modal show={showModal} onClose={handleClose}>